uk.co.compucorp.civicrm.hrcomments/api/v3/Comment.php

```diff
@@ -31,16 +31,60 @@ function civicrm_api3_comment_create($params) {
  * @throws API_Exception
  */
 function civicrm_api3_comment_delete($params) {
-  return _civicrm_api3_basic_delete(_civicrm_api3_get_BAO(__FUNCTION__), $params);
+  return _civicrm_api3_comment_soft_delete(_civicrm_api3_get_BAO(__FUNCTION__), $params);
```

Comment:
The `_civicrm_api3_basic_delete` functions call the BAO's `del` method internally. So, instead of implementing this soft delete logic on the API level, please implement on the BAO. This is also important because, if we don't want to use the API for some reason, the BAO will work the same way